packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1110 @@
+                }]
+              ]}>
+              {sortDirection != null && (
+                sortDirection === 'ascending' ? <SortUpArrow /> : <SortDownArrow />
+              )}
+            </Provider>
+          )}
+          <div className={columnHeaderText}>
+            {children}
+          </div>
+          <Chevron size="M" className={chevronIcon} />
+        </Button>
+        <Menu onAction={onMenuSelect} items={items}>
+          {(item) => <MenuItem>{item?.label}</MenuItem>}
+        </Menu>
+      </MenuTrigger>
+      <div data-react-aria-prevent-focus="true">
+        <ColumnResizer data-react-aria-prevent-focus="true" className={({resizableDirection, isResizing}) => resizerHandleContainer({resizableDirection, isResizing, isHovered: isInResizeMode || isHovered})}>
+          {({isFocusVisible, isResizing}) => (
+            <>
+              <ResizerIndicator isInResizeMode={isInResizeMode} isFocusVisible={isFocusVisible} isHovered={isHovered} isResizing={isResizing} />
+              {(isFocusVisible || isInResizeMode) && isResizing && <div className={nubbin}><Nubbin /></div>}
+            </>
+        )}
+        </ColumnResizer>
+      </div>
+    </>
+  );
+}
+
+function ResizerIndicator({isFocusVisible, isHovered, isResizing, isInResizeMode}) {
+  let state = useContext(ResizableTableContainerContext);
+
+  return (
+    <div style={{height: isResizing ? state?.tableHeight : '100%'}} className={resizerHandle({isFocusVisible, isHovered: isHovered || isInResizeMode, isResizing})} />
+  );
+}
+
+const tableHeader = style({
+  height: 'full',
+  width: 'full',
+  backgroundColor: 'gray-75'
+});
+
+const selectAllCheckbox = style({
+  marginStart: 16, // table-edge-to-content, same between mobile and desktop
+  marginEnd: {
+    default: 8,
+    scale: {
+      large: 14
+    }
+  }
+});
+
+const selectAllCheckboxColumn = style({
+  padding: 0,
+  height: '[calc(100% - 1px)]',
+  outlineStyle: 'none',
+  position: 'relative',
+  alignContent: 'center',
+  borderColor: 'gray-300',
+  borderXWidth: 0,
+  borderTopWidth: 0,
+  borderBottomWidth: 1,
+  borderStyle: 'solid',
+  backgroundColor: 'gray-75'
+});
+
+let InternalTableHeaderContext = createContext<{isHeaderRowHovered?: boolean}>({isHeaderRowHovered: false});
+
+export function TableHeader<T extends object>({columns, children}: TableHeaderProps<T>) {
+  let {scale} = useContext(InternalTableContext);
+  let {selectionBehavior, selectionMode} = useTableOptions();
+  let [isHeaderRowHovered, setHeaderRowHovered] = useState(false);
+
+  return (
+    <InternalTableHeaderContext.Provider value={{isHeaderRowHovered}}>
+      <RACTableHeader onHoverChange={setHeaderRowHovered} className={tableHeader}>
+        {/* Add extra columns for selection. */}
+        {selectionBehavior === 'toggle' && (
+          // Also isSticky prop is applied just for the layout, will decide what the RAC api should be later
+          // @ts-ignore
```

Comment:
this is just to get it on the node properties?

Reply:
yep that is correct